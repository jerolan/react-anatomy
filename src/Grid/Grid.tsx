import styled from 'styled-components';

import FlexBox from '../FlexBox';
import { GridRowColScalesProps } from '../types';

const GridBase = styled(FlexBox)<GridRowColScalesProps>`
  flex: 1;
  flex-direction: ${props => props.xs || 'column'};

  @media (min-width: 768px) {
    flex-direction: ${props => props.sm};
  }

  @media (min-width: 992px) {
    flex-direction: ${props => props.md};
  }

  @media (min-width: 1200px) {
    flex-direction: ${props => props.lg};
  }
`;

export default GridBase;
