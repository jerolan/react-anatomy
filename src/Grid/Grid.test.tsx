import React from 'react';
import renderer from 'react-test-renderer';

import Grid from './Grid';

describe('<Grid />', () => {
  it('renderenders without crashing - no props', () => {
    const component = renderer.create(<Grid />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
