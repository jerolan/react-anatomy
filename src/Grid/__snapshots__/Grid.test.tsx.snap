// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renderenders without crashing - no props 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH bmZEGC"
/>
`;
